### PR TITLE
Fix cyclic dependency

### PR DESCRIPTION
**Pull request by CC007**
Elementary depends on Utilitary, but Utilitary's tests also depended on Elementary. This cyclic dependency caused issues when building and testing. Furthermore, this test dependency referred to an older version of Elementary, which caused junit dependency mismatch issues.

All this is now solved by introducing a new itest module for all tests in Utilitary that depended on Elementary.

Signed-off-by: CC007

### CODE REVIEW

**Review by Pante — COMMENTED** [review]
Thanks for the PR! I think moving the tests into a separate module is a good way to break the cyclic dependencies. However, I don't think `itest`/integration tests is a good way to describe the tests. They are unit tests for utilitary after all.

**Comment by CC007** [comment]
They aren't really unit tests. Though, because you aren't really testing the unit by itself (which would be hard given the limitations of compiler testing). You're testing it's use when integrated with the elementary ToolsExtension.

**Comment by Pante** [comment]
Hmm fair enough

File: utilitary-itest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>elementary-project</artifactId>
+        <groupId>com.karuslabs</groupId>
+        <version>1.2.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>utilitary-itest</artifactId>
+    <packaging>jar</packaging>
+    <name>Utilitary integration test</name>
+    <description>The integration test module for testing the Utilitary module.</description>
```
Comment (Pante) [inline utilitary-itest/pom.xml:14]: Nitpick: I think the description can be better phrased as "Integration tests for Utilitary."

File: utilitary-itest/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>elementary-project</artifactId>
+        <groupId>com.karuslabs</groupId>
+        <version>1.2.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>utilitary-itest</artifactId>
+    <packaging>jar</packaging>
+    <name>Utilitary integration test</name>
```
Comment (Pante) [inline utilitary-itest/pom.xml:13]: Nit[ick: I think it should be "Utilitary Integration Tests".

**Review by Pante — CHANGES_REQUESTED** [review]
LGTM, just two minor nitpicks. Thanks for taking the time to submit a PR!

File: utilitary-itest/pom.xml (hunk 2 above)
Comment (CC007) [inline utilitary-itest/pom.xml:13]: @Pante you can also add suggestions like so:
```suggestion
    <name>Utilitary Integration Tests</name>
```

File: utilitary-itest/pom.xml (hunk 1 above)
Comment (CC007) [inline utilitary-itest/pom.xml:14]:
```suggestion
    <description>Integration tests for Utilitary.</description>
```

File: utilitary-itest/pom.xml (hunk 2 above)
Comment (Pante) [reply utilitary-itest/pom.xml]: Ohhh, that’s really neat, I never knew that feature existed.

**Review by Pante — APPROVED** [review]

**Comment by Pante** [comment]
Yup, I’m going to disable codecov altogether soon

**Pull request merged**